### PR TITLE
Targets for channel 1 and 2; different QuantaSoft export format

### PR DESCRIPTION
**Pull request by acnb**
Hi Dean,

people in my lab often perform several experiments on the same ddpcr plate. To use your package with these plates, I wrote some code to import target information into plate_meta. I've noticed, that our export files look slighly different from yours. 

Your column names are: 
Well,ExptType,Experiment,Sample,TypeAssay,Assay,...
Ours are:
Well,ExptType,Experiment,Sample,TargetType,Target,...

We are using QuantaSoft version 1.7.4 . I did unfortunately not find any version history from QuantaSoft.

My code should be able to handle both formats. 

Please advice on my pull request.

Thanks,

Andreas

### CODE REVIEW

File: R/plate-utils.R [hunk 1]
```diff
@@ -77,12 +77,32 @@ init_meta <- function(plate) {
   if (is.null(plate_meta)) {
     plate_meta <- DEFAULT_PLATE_META
   } else {
-    meta_cols_keep <- c("well", "sample")
+    meta_cols_keep <- c("well", "sample", "target_ch1", "target_ch2")
     tryCatch({
+      
+      if (colnames(plate_meta)[5] == 'typeassay' |
```
Comment (daattali) [inline R/plate-utils.R]: Does the number 5 have to be hardcoded or would it be ok to just see if this variable exists at all in any of the columns?

File: R/plate-utils.R [hunk 2]
```diff
@@ -77,12 +77,32 @@ init_meta <- function(plate) {
...
   } else {
-    meta_cols_keep <- c("well", "sample")
+    meta_cols_keep <- c("well", "sample", "target_ch1", "target_ch2")
     tryCatch({
+      
+      if (colnames(plate_meta)[5] == 'typeassay' | 
+          colnames(plate_meta)[5] == 'targettype'){
+        # different versions of QuantaSoft
+        if (colnames(plate_meta)[5] == 'typeassay'){
+          plate_meta %<>% dplyr::mutate(target = assay,
```
Comment (daattali) [inline R/plate-utils.R]: Could you try to use the NSE version `mutate_` (same for `group_by_` and `summarise_` below)

File: R/subset.R [hunk 3]
```diff
@@ -58,14 +61,36 @@ subset.ddpcr_plate <- function(x, wells, samples, ...) {
       plate_meta(x) %>%
       dplyr::filter_(~ sample %in% samples) %>%
       .[['well']]
+  } else if (!missing(targets_ch1) | !missing(targets_ch2)) {
```
Comment (daattali) [inline R/subset.R]: this should be a double `||` I believe?

File: R/subset.R [hunk 4]
```diff
@@ -58,14 +61,36 @@ subset.ddpcr_plate <- function(x, wells, samples, ...) {
...
+      plate_meta(x) %>%
+      .[['well']]
   } else {
     return(x)  # if no arguments, just return the same plate
   }
   
+  if (!missing(targets_ch1)) {
+    wells_from_targets <-
+      plate_meta(x) %>%
+      dplyr::filter_(~ target_ch1 %in% targets_ch1) %>%
```
Comment (daattali) [inline R/subset.R]: Does this code work ok without error if the user gives targets that are not actually in the data?

File: tests/testthat/test-subset.R [hunk 5]
```diff
@@ -102,3 +102,11 @@ test_that("get_col works", {
   expect_identical(get_col("C05"), "05")
   expect_identical(get_col("D11"), "11")
 })
+
+test_that("subset targets works", {
+  dir <- system.file("sample_data", "read_v174", package = "ddpcr")
+  plate <- new_plate(dir)
+  expect_identical(plate %>% subset(targets_ch1 = "t1.fw") %>%
+                     wells_used,
```
Comment (daattali) [inline tests/testthat/test-subset.R]: can you sort the indendation here to be consistent (I think the `wells_used` could go on the previous line)

File: R/plate-utils.R (hunk 2 above)
Comment (acnb) [inline R/plate-utils.R]: Ok, I'll try.

File: R/subset.R (hunk 3 above)
Comment (acnb) [inline R/subset.R]: You're right, I'll change that.

File: R/subset.R (hunk 4 above)
Comment (acnb) [inline R/subset.R]: If invalid targets are given, the plate is returned unchanged. I did this to stay consistent with previous behavior. subset(plate, samples = 'noSample') returns plate as well.  However, I'm not sure, if this is the best way. Sample and targets are usually some abbreviations, that are prone to typing errors.  At warning might make analysis safer, e.g.:

    # if no valid wells were given, don't do anything
    if (sum(wells %in% (x %>% wells_used)) == 0) {
       warn_msg("No valid wells selected. Returning plate unchanged.")
       return(x)
    }

File: tests/testthat/test-subset.R (hunk 5 above)
Comment (acnb) [inline tests/testthat/test-subset.R]: ok.

File: R/subset.R (hunk 4 above)
Comment (daattali) [inline R/subset.R]: you're right, I agree! Do you mind adding warnings?

Comment (acnb) [inline R/subset.R]: Certainly!

**Comment by daattali** [comment]
Thanks for the very high quality PR

**Comment by daattali** [comment]
@acnb one last request actually: could you add a bullet point in the NEWS (or just give me the text of what should be added)

**Comment by acnb** [comment]
Your news bullets point looks perfect, I have nothing to add.

**Pull request merged**